### PR TITLE
New deployment targets for cloud dashboard in STAGING

### PR DESCRIPTION
**Pull request by magul**
(Will require cherry-picking of deploy-staging.yaml and prepare-release.yaml  to release branches)

Relates saleor/saleor-cloud-infra#1029

### CODE REVIEW

**Comment by krzysztofzuraw** [comment]
The secrets have been set.

**Review by NyanKiyoshi — CHANGES_REQUESTED** [review]
Follow https://docs.github.com/en/actions/security-guides/security-hardening-for-github-actions#understanding-the-risk-of-script-injections

File: .github/workflows/deploy-master-staging.yaml [hunk 1]
```diff
@@ -43,21 +43,32 @@ jobs:
...
+          aws-secret-access-key: ${{ secrets.AWS_CLOUD_STAGING_SECRET_ACCESS_KEY }}
           aws-region: us-east-1
 
       - name: Deploy master staging
         run: |
-          aws s3 sync build/dashboard s3://${{ secrets.AWS_STAGING_DEPLOYMENT_BUCKET }}/saleor-master-staging/static/
-          aws s3 cp build/dashboard/index.html s3://${{ secrets.AWS_STAGING_DEPLOYMENT_BUCKET }}/saleor-master-staging/
-          for i in {1..3}; do aws cloudfront create-invalidation --distribution-id ${{ secrets.AWS_STAGING_CF_DIST_ID }} --paths "/dashboard*" && break || sleep 5; done
+          aws s3 sync build/dashboard s3://${DEPLOYMENT_BUCKET}/saleor-master-staging/static/
+          aws s3 cp build/dashboard/index.html s3://${DEPLOYMENT_BUCKET}/saleor-master-staging/
```
Comment (NyanKiyoshi) [inline .github/workflows/deploy-master-staging.yaml:53]:
```suggestion
          aws s3 sync build/dashboard "s3://${DEPLOYMENT_BUCKET}/saleor-master-staging/static/"
          aws s3 cp build/dashboard/index.html "s3://${DEPLOYMENT_BUCKET}/saleor-master-staging/"
```

File: .github/workflows/deploy-master-staging.yaml [hunk 2]
```diff
@@ -43,21 +43,32 @@ jobs:
...
+        env:
+          DEPLOYMENT_BUCKET: ${{ secrets.AWS_CLOUD_STAGING_DEPLOYMENT_BUCKET }}
 
       - name: Deploy fake production
         run: |
-          aws s3 sync build/dashboard s3://${{ secrets.AWS_STAGING_DEPLOYMENT_BUCKET }}/saleor-fake-production/static/
-          aws s3 cp build/dashboard/index.html s3://${{ secrets.AWS_STAGING_DEPLOYMENT_BUCKET }}/saleor-fake-production/
-          for i in {1..3}; do aws cloudfront create-invalidation --distribution-id ${{ secrets.AWS_STAGING_CF_DIST_ID }} --paths "/dashboard*" && break || sleep 5; done
+          aws s3 sync build/dashboard s3://${DEPLOYMENT_BUCKET}/saleor-fake-production/static/
+          aws s3 cp build/dashboard/index.html s3://${DEPLOYMENT_BUCKET}/saleor-fake-production/
```
Comment (NyanKiyoshi) [inline .github/workflows/deploy-master-staging.yaml:60]:
```suggestion
          aws s3 sync build/dashboard "s3://${DEPLOYMENT_BUCKET}/saleor-fake-production/static/"
          aws s3 cp build/dashboard/index.html "s3://${DEPLOYMENT_BUCKET}/saleor-fake-production/"
```

File: .github/workflows/deploy-master-staging.yaml [hunk 3]
```diff
@@ -43,21 +43,32 @@ jobs:
...
+        env:
+          DEPLOYMENT_BUCKET: ${{ secrets.AWS_CLOUD_STAGING_DEPLOYMENT_BUCKET }}
+
+      - name: Invalidate cache
+        run: |
+          IFS=',' read -ra DISTRIBUTIONS_ARRAY <<< "$CDN_DISTRIBUTIONS"
+          for DISTRIBUTION in "${DISTRIBUTIONS_ARRAY[@]}"; do
+            for i in {1..3}; do aws cloudfront create-invalidation --distribution-id $DISTRIBUTION --paths "/dashboard*" && break || sleep 5; done
```
Comment (NyanKiyoshi) [inline .github/workflows/deploy-master-staging.yaml:68]:
```suggestion
            for i in {1..3}; do aws cloudfront create-invalidation --distribution-id "$DISTRIBUTION" --paths "/dashboard*" && break || sleep 5; done
```

File: .github/workflows/deploy-staging.yaml [hunk 4]
```diff
@@ -70,15 +70,22 @@ jobs:
...
+          aws-secret-access-key: ${{ secrets.AWS_CLOUD_STAGING_SECRET_ACCESS_KEY }}
           aws-region: us-east-1
 
       - name: Deploy
         run: |
-          aws s3 sync build/dashboard s3://${{ secrets.AWS_STAGING_DEPLOYMENT_BUCKET }}/${ENVIRONMENT}/static/
-          aws s3 cp build/dashboard/index.html s3://${{ secrets.AWS_STAGING_DEPLOYMENT_BUCKET }}/${ENVIRONMENT}/
-          for i in {1..3}; do aws cloudfront create-invalidation --distribution-id ${{ secrets.AWS_STAGING_CF_DIST_ID }} --paths "/dashboard*" && break || sleep 5; done
+          aws s3 sync build/dashboard s3://${DEPLOYMENT_BUCKET}/${ENVIRONMENT}/static/
+          aws s3 cp build/dashboard/index.html s3://${DEPLOYMENT_BUCKET}/${ENVIRONMENT}/
```
Comment (NyanKiyoshi) [inline .github/workflows/deploy-staging.yaml:80]:
```suggestion
          aws s3 sync build/dashboard "s3://${DEPLOYMENT_BUCKET}/${ENVIRONMENT}/static/"
          aws s3 cp build/dashboard/index.html "s3://${DEPLOYMENT_BUCKET}/${ENVIRONMENT}/"
```

File: .github/workflows/deploy-staging.yaml [hunk 5]
```diff
@@ -70,15 +70,22 @@ jobs:
...
+
+          IFS=',' read -ra DISTRIBUTIONS_ARRAY <<< "${CDN_DISTRIBUTIONS}"
+          for DISTRIBUTION in "${DISTRIBUTIONS_ARRAY[@]}"; do
+            for i in {1..3}; do aws cloudfront create-invalidation --distribution-id $DISTRIBUTION --paths "/dashboard*" && break || sleep 5; done
```
Comment (NyanKiyoshi) [inline .github/workflows/deploy-staging.yaml:84]:
```suggestion
            for i in {1..3}; do aws cloudfront create-invalidation --distribution-id "$DISTRIBUTION" --paths "/dashboard*" && break || sleep 5; done
```

File: .github/workflows/prepare-release.yml [hunk 6]
```diff
@@ -68,14 +68,22 @@ jobs:
...
+          aws-access-key-id: ${{ secrets.AWS_CLOUD_STAGING_ACCESS_KEY_ID }}
+          aws-secret-access-key: ${{ secrets.AWS_CLOUD_STAGING_SECRET_ACCESS_KEY }}
           aws-region: us-east-1
       - name: Deploy
         run: |
-          aws s3 sync build/dashboard s3://${{ secrets.AWS_STAGING_DEPLOYMENT_BUCKET }}/${ENVIRONMENT}/static/
-          aws s3 cp build/dashboard/index.html s3://${{ secrets.AWS_STAGING_DEPLOYMENT_BUCKET }}/${ENVIRONMENT}/
-          for i in {1..3}; do aws cloudfront create-invalidation --distribution-id ${{ secrets.AWS_STAGING_CF_DIST_ID }} --paths "/dashboard*" && break || sleep 5; done
+          aws s3 sync build/dashboard s3://${DEPLOYMENT_BUCKET}/${ENVIRONMENT}/static/
+          aws s3 cp build/dashboard/index.html s3://${DEPLOYMENT_BUCKET}/${ENVIRONMENT}/
```
Comment (NyanKiyoshi) [inline .github/workflows/prepare-release.yml:77]:
```suggestion
          aws s3 sync build/dashboard "s3://${DEPLOYMENT_BUCKET}/${ENVIRONMENT}/static/"
          aws s3 cp build/dashboard/index.html "s3://${DEPLOYMENT_BUCKET}/${ENVIRONMENT}/"
```

File: .github/workflows/prepare-release.yml [hunk 7]
```diff
@@ -68,14 +68,22 @@ jobs:
...
+
+          IFS=',' read -ra DISTRIBUTIONS_ARRAY <<< "${CDN_DISTRIBUTIONS}"
+          for DISTRIBUTION in "${DISTRIBUTIONS_ARRAY[@]}"; do
+            for i in {1..3}; do aws cloudfront create-invalidation --distribution-id $DISTRIBUTION --paths "/dashboard*" && break || sleep 5; done
```
Comment (NyanKiyoshi) [inline .github/workflows/prepare-release.yml:81]:
```suggestion
            for i in {1..3}; do aws cloudfront create-invalidation --distribution-id "$DISTRIBUTION" --paths "/dashboard*" && break || sleep 5; done
```

**Review by NyanKiyoshi — DISMISSED** [review]

File: .changeset/spicy-carrots-move.md
Comment (NyanKiyoshi) [inline .changeset/spicy-carrots-move.md:1]: Is changelog entry really necessary for internal things? It doesn't bring anything to the community, it's most likely noise in changelogs

Comment (magul) [reply .changeset/spicy-carrots-move.md:1]: I was not able to merge without it, basically one check was failing :/

**Review by andrzejewsky — APPROVED** [review]

File: .changeset/spicy-carrots-move.md
Comment (NyanKiyoshi) [reply .changeset/spicy-carrots-move.md:1]: @saleor/dashboard?

**Review by andrzejewsky — DISMISSED** [review]

**Review by poulch — DISMISSED** [review]

File: .changeset/spicy-carrots-move.md
Comment (andrzejewsky) [reply .changeset/spicy-carrots-move.md:1]: yes it is. We not use these logs just for community but also for preparing releases. Based on that, another action creates pull request on each release that contains all changes delivered in the given release and workflow changes are also the work that was done

File: .github/workflows/deploy-master-staging.yaml [hunk 8]
```diff
@@ -43,21 +43,32 @@ jobs:
...
+          DEPLOYMENT_BUCKET: ${{ secrets.AWS_CLOUD_STAGING_DEPLOYMENT_BUCKET }}
+
+      - name: Invalidate cache
+        run: |
+          IFS=',' read -ra DISTRIBUTIONS_ARRAY <<< "$CDN_DISTRIBUTIONS"
+          for DISTRIBUTION in "${DISTRIBUTIONS_ARRAY[@]}"; do
+            for i in {1..3}; do aws cloudfront create-invalidation --distribution-id "$DISTRIBUTION" --paths "/dashboard*" && break || sleep 5; done
+          done
+        env:
+          CDN_DISTRIBUTIONS: ${{ secrets.AWS_CLOUD_STAGING_CDN_DISTRIBUTIONS }}
```
Comment (NyanKiyoshi) [inline .github/workflows/deploy-master-staging.yaml:71]: Is `CDN_DISTRIBUTIONS` a secret or not? Secrets shouldn't mutate as they also are removed from logs if found (you are splitting the value)

**Review by NyanKiyoshi — CHANGES_REQUESTED** [review]

File: .changeset/spicy-carrots-move.md
Comment (andrzejewsky) [reply .changeset/spicy-carrots-move.md:1]: it we all think it's useless, perhaps we can exclude `.github` folder somehow. However I think during releases worth to mention all changes, they are also work that was delivered + it's one file, there is even a link that adds it automatically

File: .github/workflows/deploy-master-staging.yaml (hunk 8 above)
Comment (magul) [reply .github/workflows/deploy-master-staging.yaml:71]: I've added
```
            echo "::add-mask::$DISTRIBUTION"
```
everywhere, where I split list of distributions.

**Review by andrzejewsky — APPROVED** [review]

**Review by NyanKiyoshi — APPROVED** [review]

**Pull request merged**